lib/tasks/translate_collections.rake

```diff
@@ -0,0 +1,277 @@
+require 'libxml_to_hash'
+
+namespace :collection do
+  desc 'Translate a collection from native format to UMM JSON and back to native format'
+  task :translate, [:file, :format, :disp, :version] => :environment do |_task, args|
+    args.with_defaults(:version => '1.15.3')
+    args.with_defaults(:disp => 'show')
+
+    puts 'FORMAT INVALID' unless args.format == ('echo10' || 'dif10' || 'iso19115')
+
+    filename = args.file.split('/')[-1]
+    puts "\nTranslating #{filename} to UMM JSON..."
+
+    native_original_xml = File.read(args.file)
+    native_original_hash = Hash.from_xml(native_original_xml)
+
+    #translate to UMM
+    umm = cmr_client.translate_collection(native_original_xml, "application/#{args.format}+xml", "application/vnd.nasa.cmr.umm+json;version=#{args.version}", skip_validation=true )
+    umm_json = umm.body.to_json
+    umm.success? ? puts("\nsuccessful translation to UMM") : puts("\nUMM translation failure")
+
+    # translate back to native
+    back_to_native = cmr_client.translate_collection(umm_json, "application/vnd.nasa.cmr.umm+json;version=#{args.version}", "application/#{args.format}+xml", skip_validation=true )
+    native_converted_hash = Hash.from_xml(back_to_native.body)
+    native_converted_xml = back_to_native.body
+    back_to_native.success? ? puts("successful translation to native format \n\n") : puts("Native format translation failure \n\n")
+
+    # nokogiri output
+    o = Nokogiri::XML(native_original_xml) { |config| config.strict.noblanks }
+    c = Nokogiri::XML(native_converted_xml) { |config| config.strict.noblanks }
+    # puts args.display, args.display.class
+      o.diff(c, {:added => true, :removed => true}) do |change,node|
+        next if path_leads_to_list?(node.parent.path, native_original_hash, native_converted_hash)
```

Comment:
Aren't all the things that trigger this list items?  Is something missing here?  It seems like you could create your list of lists to check from this data.

Reply:
I'm a little unsure of your question. If I understand correctly, you're asking 'why can't you use the nokogiri output to find all the paths that lead to array changes instead of writing custom code to find those paths again?' 

If that is your question, I think you make a great point, however, I believe I looked into that originally, but found that nokogiri wasn't recognizing some changes at all or, further, wasn't generating useful paths for some list additions/removals. But I will certainly look at it again to see if there is optimization potential. **Please confirm if I understood/misunderstood your question**

Reply:
You understood the question correctly.